torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -777,13 +777,21 @@ kir::ExpressionEvaluator bindKernelInputs(
           "Something went wrong configuring launch. Inputs no longer match.");
 
       for (const auto dim : c10::irange(root_domain.size())) {
-        Val* extent = nullptr;
+        const auto extent = root_domain[dim]->extent();
         if (root_domain[dim]->hasExpandedExtent()) {
-          extent = root_domain[dim]->expandedExtent();
-        } else {
-          extent = root_domain[dim]->extent();
+          TORCH_INTERNAL_ASSERT(
+              aten_tensor.strides()[dim] == 0,
+              "Execting an expanded dimension on ",
+              inputs[i]->toString(),
+              " dimension ",
+              dim,
+              " but found stride ",
+              aten_tensor.strides()[dim]);
         }
-        const auto value = aten_tensor.sizes()[dim];
+
+        const auto value = root_domain[dim]->hasExpandedExtent()
+            ? 1
+            : aten_tensor.sizes()[dim];
```

Comment:
We shouldn't actually expand a tensor until we have to. Until it's resolved expand is just a broadcast. Expand is really just a WAR that when we don't actually resolve the broadcast from an expanded dim, we still have to return its expanded size instead of 1.

Reply:
We want the expanded extent to be 1 as we just consider it a broadcast dim, not a physical dim. So here we just treat it like a broadcast dim.